test/cat.jl

```diff
@@ -51,23 +57,30 @@ module TestCat
 
     @testset "hcat ::AbstractVectors" begin
         df = DataFrame()
-        DataFrames.hcat!(df, CategoricalVector{Union{Int, Missing}}(1:10))
+        global df2 = DataFrame(x1=CategoricalVector{Union{Int,Missing}}(1:10))
```

Comment:
Is global causing a problem?  I would prefer to just rename everything instead of living with the warning if that's what it takes.  The warnings are more than just annoying, they make it very difficult to spot real problems, especially in 0.7.0.

The original `hcat` functions were causing deprecation warnings.  Might be fixed now with alternate naming, let me try changing it back.

Reply:
Julia 0.7 will go away in a few months, but the code base is going to stay for years. So I'd rather not add incorrect statements just to silence warnings. Clearly `df2` isn't global here. It's fine to use this temporarily to help finding real warnings, but better not include it in the final PR.

Reply:
Ok, I'm getting rid of all of the `global`s now.  I'm renaming all of the variables so that they don't produce warnings.